### DISCUSSION TITLE
decode jwt with rs256 algorithm in golang-jwt / jwt

### DISCUSSION THREAD

**Discussion by chaotic98** [golang-jwt/jwt#292] [Q&A] [answered]
the documentation only has parsing and validate with Hmac 
https://pkg.go.dev/github.com/golang-jwt/jwt/v5#example-Parse-Hmac
is there any way to decode in rs256 ?

**Comment by dfath** [comment]
Try using ParseUnverified, it useful if you want to extract values without validate.

https://github.com/golang-jwt/jwt/blob/148d71010923ca691c950db9846191800f498f8d/parser_test.go#L430

**Comment by oxisto** [comment, marked as answer]
Please do NOT use `ParseUnverified` unless you ***REALLY*** know what you are doing. 

We really need to supply an appropriate example using asymmetric keys. Basically it works the same way as for HMAC, but instead of supplying a `[]byte` key, you need to supply a `*rsa.PublicKey` in the keyfunc. Something like

```go
var myPublicKey *rsa.PublicKey

// Load key from file
myPublicKey = /*...*/

token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
	// Don't forget to validate the alg is what you expect:
	if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
		return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
	}

	return myPublicKey, nil
})
```
